pkgs/development/compilers/rust/rustc.nix

```diff
@@ -125,6 +125,19 @@ in stdenv.mkDerivation rec {
 
     # Useful debugging parameter
     # export VERBOSE=1
+  '' +
+  # From Arch/Debian: prevent SSE2 instructions in i686 output.
+  # While the rest of the userspace is built for the P6 instruction set
+  # architecture (Pentium Pro and later), Rust arbitrarily redefines i686 to
+  # include SSE2 instructions as introduced by the P68 instruction set
+  # architecture (Pentium 4 and later). To be able to run NixOS i686-linux
+  # on all 686 processors, we have to tell Rust to exclude SSE2 instructions,
+  # as GCC does for the rest of userspace.
+  optionalString stdenv.targetPlatform.isx86_32 ''
+    substituteInPlace vendor/rustc-ap-rustc_target/src/spec/i686_unknown_linux_gnu.rs \
+      --replace "pentium4" "pentiumpro"
+    substituteInPlace compiler/rustc_target/src/spec/i686_unknown_linux_gnu.rs \
+      --replace "pentium4" "pentiumpro"
```

Comment:
I would apply the patches unconditionally to avoid bitrot and also make rustc cross-compile correctly.
Ideally a patch file instead of `substituteInPlace` to make the build on upgrade fail otherwise....

Reply:
or `optionalString (!stdenv.targetPlatform.sse2Support) ''`

there is no `targetPlatform.sse2Support`, but there is `targetPlatform.sse3Support` splitting `x86_64` processors to those who support SSE3 and who does not.

Adding `targetPlatform.sse2Support` to split `x86_32` processors in the same way seems natural